src/libsync/creds/httpcredentials.cpp

```diff
@@ -300,6 +300,9 @@ bool HttpCredentials::refreshAccessTokenInternal(int tokenRefreshRetriesCount)
         case QNetworkReply::OperationCanceledError:
             [[fallthrough]];
         case QNetworkReply::TemporaryNetworkFailureError:
+            [[fallthrough]];
+        // VPN not ready?
+        case QNetworkReply::ConnectionRefusedError:
```

Comment:
maybe we should invert the logic and only invalidate for special cases?

Reply:
Sounds reasonable